### PR TITLE
fix(exec): Refactor binarySource out to global setting

### PR DESCRIPTION
**Pull request by zharinov**
ref #2863

### CODE REVIEW

File: lib/constants/data-binary-source.ts [hunk 1]
```diff
@@ -23,6 +23,11 @@ export const DATASOURCE_TERRAFORM = 'terraform';
 export const DATASOURCE_TERRAFORM_PROVIDER = 'terraformProvider';
 
 // Binary Source
+export enum BinarySource {
+  Auto = 'auto',
+  Docker = 'docker',
+  Global = 'global',
+}
```
Comment (zharinov) [inline lib/constants/data-binary-source.ts]: Here is the decision point: (1) we can not use values in enums (even `const`'s), but (2) we want to refer particular values through literals so can't use use sum types like `type X = 'a' | 'b'`, and (3) we want more type guarantees.

So, do I need change constants below to the enum? (ref #5088)

Comment (rarkins) [reply lib/constants/data-binary-source.ts]: I don't have enough experience with TypeScript to be sure, so let's just do something that's an improvement and iteratively improve it later. Also to note:

1. This is a setting by the "sysadmin" of a Renovate installation and should be defined in just one place. Such n option is much less important than enums that might occur in multiple areas of the app.

2. The most important place to be enforcing is probably in https://github.com/renovatebot/renovate/blob/d871b4c8c908a8e96ed62eefcc9e50b19d321142/lib/config/definitions.ts#L192-L199 and we're not doing it yet. So any eventual solution probably needs to be based off an `allowedValues` array in that definitions file.

Comment (rarkins) [reply lib/constants/data-binary-source.ts]: Seems like nothing to lose by changing the below to be `BinarySource.Auto` etc?

File: lib/manager/poetry/artifacts.ts [hunk 2]
```diff
@@ -34,23 +33,15 @@ export async function updateArtifacts({
...
-    }
     const cmd: string[] = [];
     for (let i = 0; i < updatedDeps.length; i += 1) {
       const dep = updatedDeps[i];
       cmd.push(`poetry update --lock --no-interaction ${dep}`);
     }
+    const execOptions: ExecOptions = {
+      cwd: join(config.localDir, subDirectory),
+      docker: { image: 'renovate/poetry' },
+    };
```
Comment (rarkins) [inline lib/manager/poetry/artifacts.ts]: This is a nice simplification

File: lib/constants/data-binary-source.ts (hunk 1 above)
Comment (zharinov) [reply lib/constants/data-binary-source.ts]: Yes, just need to replace all constants to enum values

**Comment by zharinov** [comment]
Ready for review

**Pull request merged**